spec/api_config_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'spec_helper'
+
+describe Quandl::ApiConfig do
+  context 'threading' do
+    context '.api_key' do
+      let(:threads) { [] }
+      let(:default_key) { (0...8).map { (65 + rand(26)).chr }.join }
+
+      it 'should allows for a default value' do
+        Quandl::ApiConfig.api_key = default_key
+
+        expect do
+          3.times do
+            threads << Thread.new do
+              expect(Quandl::ApiConfig.api_key).to eq(default_key)
+
+              api_key = (0...8).map { (65 + rand(26)).chr }.join
+              Quandl::ApiConfig.api_key = api_key
+              expect(Quandl::ApiConfig.api_key).to eq(api_key)
+            end
+            sleep(1) # Important to ensure that previous thread code has executed
+          end
+
+          threads.map(&:join)
+        end.not_to change {
+          Quandl::ApiConfig.api_key
+        }
+      end
+
+      it 'allows users to change their settings per thread' do
+        3.times do
+          threads << Thread.new do
+            api_key = (0...8).map { (65 + rand(26)).chr }.join
+            Quandl::ApiConfig.api_key = api_key
+            expect(Quandl::ApiConfig.api_key).to eq(api_key)
```

Comment:
maybe put an expectation that `Quandl::ApiConfig.api_key` should be nil by default in a new thread?